File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-14.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) == 0:
            return 0
        longest = 1
        l, r = 0, 1
        while r < len(s):
            if s[r] not in set(s[l:r]):
                current_len = len(s[l:r+1])
                r += 1
            else:
                current_len = len(s[l:r])
                l += 1
            longest = max(longest, current_len)
        return longest

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-14.rs
impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        let mut mp: HashSet<_> = nums.iter().copied().collect();
        let mut longest = 0;
        for &num in &mp {
            if !mp.contains(&(num - 1)) {
                let mut length = 1;
                while mp.contains(&(num + length)) {
                    length += 1;
                }
                longest = max(length, longest);
            }
        }
        longest
    }
}
